packages/console/src/pages/ApplicationDetails/components/GuideDrawer/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { type Application } from '@logto/schemas';
+import { useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import ArrowLeft from '@/assets/icons/arrow-left.svg';
+import Close from '@/assets/icons/close.svg';
+import IconButton from '@/ds-components/IconButton';
+import Spacer from '@/ds-components/Spacer';
+import { type SelectedGuide } from '@/pages/Applications/components/GuideCard';
+import GuideGroup from '@/pages/Applications/components/GuideGroup';
+import useAppGuideMetadata from '@/pages/Applications/components/GuideLibrary/hook';
+import GuideV2 from '@/pages/Applications/components/GuideV2';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  app: Application;
+  onClose: () => void;
+};
+
+function GuideDrawer({ app, onClose }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console.applications.guide' });
+  const [_, getStructuredMetadata] = useAppGuideMetadata();
```

Comment:
better to add clear comments on what is `getStructuredMetadata` and `getFilteredMetadata`. also they can return as an object to avoid explicit `_`

Reply:
This hook needs to be refactored. Me myself also find it weird tonight. Will improve later this week